core/persistence-jpa/src/main/java/org/apache/syncope/core/persistence/jpa/entity/user/JPALinkedAccount.java

```diff
@@ -157,10 +166,9 @@ public void setEncodedPassword(final String password, final CipherAlgorithm ciph
     }
 
     @Override
-    public void setPassword(final String password, final CipherAlgorithm cipherAlgoritm) {
+    public void setPassword(final String password) {
         try {
-            this.password = ENCRYPTOR.encode(password, cipherAlgoritm);
-            this.cipherAlgorithm = cipherAlgoritm;
+            this.password = ENCRYPTOR.encode(password, this.cipherAlgorithm);
```

Comment:
What happens if `this.cipherAlgorithm` is `null`? it should take the default value from conf IIRC

Reply:
We already do: https://github.com/apache/syncope/blob/2_1_X/core/persistence-jpa/src/main/java/org/apache/syncope/core/persistence/jpa/entity/user/JPAUser.java#L194